url/src/lib.rs

```diff
@@ -2143,6 +2143,23 @@ impl Url {
         self.host_end = to_u32(self.serialization.len()).unwrap();
         self.host = host.into();
 
+        // Adjust serialization to switch between host and empty segment
+        if suffix.starts_with("/.//") {
+            suffix = suffix["/.".len()..].to_string();
```

Comment:
```suggestion
            suffix.drain(.."/.".len());
```

I think this should avoid an unnecessary allocation.